### PR TITLE
feat(Switch, SwitchButton): fix a11y issues

### PR DESCRIPTION
**Pull request by QuiiBz**
## Type

- Feature

#### The following changes where made:

Fix a11y issues with `Switch` and `SwitchButton`.

- Rename `aria-checked` to `data-checked`
- Not forward not needed props
- Replace `<label />` in `Switch` by a `<div />` and add `aria-label` prop to the `<input />`, since the switch don't have a visible label

### CODE REVIEW

File: src/components/Switch/index.tsx [hunk 1]
```diff
@@ -287,6 +288,7 @@ const Switch = ({
           </SwitchBallContainer>
           <StyledCheckbox
             id={id || name}
+            aria-label={`${name}-label`}
```
Comment (chambo-e) [inline src/components/Switch/index.tsx:291]: accessibility wise this is worst than having nothing, we are ticking a11y checks with placeholder data

Comment (chambo-e) [reply src/components/Switch/index.tsx:291]: Could we force either aria-label or label instead ?

**Review by chambo-e — CHANGES_REQUESTED** [review]

File: src/components/Switch/index.tsx (hunk 1 above)
Comment (QuiiBz) [reply src/components/Switch/index.tsx:291]: Label needs real content, we don't always have something to write as a label. Maybe we should add a new `label` prop and set this `aria-label={label}`? cc @matthprost

Comment (matthprost) [reply src/components/Switch/index.tsx:291]: @QuiiBz Is there an issue of using name and put it into `aria-label={name}` ? If there is then yes I agree you can create a prop `label` and set `aria-label={label}`.

File: src/components/Switch/index.tsx [hunk 2]
```diff
@@ -273,7 +274,7 @@ const Switch = ({
         ) : null}
         <StyledSwitch
           size={size}
-          aria-checked={checked}
+          data-checked={checked}
```
Comment (matthprost) [inline src/components/Switch/index.tsx:277]: Can't we use `checked={}` directly ?

**Review by matthprost — CHANGES_REQUESTED** [review]

File: src/components/Switch/index.tsx (hunk 2 above)
Comment (QuiiBz) [reply src/components/Switch/index.tsx:277]: `StyledSwitch` is not an input but a div

**Review by matthprost — APPROVED** [review]

**Pull request merged**